test/idToken.tests.js

```diff
@@ -0,0 +1,308 @@
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when the audience is not a string or array', function(done) {
+    var EXPECTED_ERROR_MESSAGE =
+      'Audience (aud) claim must be a string or array of strings present in the ID token';
+    try {
+      idToken.validate(generateJWT({ aud: undefined }), defaultOptions);
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when expected audience doesnt match claim audience', function(done) {
+    var EXPECTED_ERROR_MESSAGE =
+      'Audience (aud) claim mismatch in the ID token; expected "expectedAudience" but was not one of "tokens-test-123, external-test-999"';
+    try {
+      expectedOptions.audience = 'expectedAudience';
+      idToken.validate(generateJWT({}), expectedOptions);
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when azp claim not found when aud has multiple values', function(done) {
+    var EXPECTED_ERROR_MESSAGE =
+      'Authorized Party (azp) claim must be a string present in the ID token when Audience (aud) claim has multiple values';
+    try {
+      var token = generateJWT({ azp: undefined });
+      idToken.validate(token, expectedOptions);
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when azp claim doesnt match the expected aud', function(done) {
+    var EXPECTED_ERROR_MESSAGE =
+      'Authorized Party (azp) claim mismatch in the ID token; expected "external-test-999", found "tokens-test-123"';
+    try {
+      var token = generateJWT({});
+      expectedOptions.audience = expectations.clientIdAlt;
+      idToken.validate(token, expectedOptions);
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+
+  it('should throw when nonce is in options, but missing from claim', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'Nonce (nonce) claim must be a string present in the ID token';
+    try {
+      var token = generateJWT({ nonce: undefined });
+
+      idToken.validate(token, expectedOptions);
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when nonce claim doesnt match nonce expected', function(done) {
```

Comment:
Do we also need a test here for when `nonce` is not a string? It's a test I've seen in other SDKs and there are similar tests in this test suite (`exp`, for example).